tools/doc/generate.js

```diff
@@ -48,11 +50,15 @@ function next(er, input) {
       break;
 
     case 'html':
-      require('./html.js').toHTML(input, inputFile, template, nodeVersion,
-        function(er, html) {
-          if (er) throw er;
-          console.log(html);
+      toHTML(input, inputFile, template, nodeVersion, (er, html) => {
+        if (er) throw er;
+        const filename = `./out/doc/api/${path.basename(inputFile, 'md')}html`;
```

Comment:
1. it can with that. basically you define the behaviour of the guide, html etc. rendering within that switch case. It would of course be possible to have a flag for the output in the makefile. A flag okay, having unix control flow through piping not so okay - especially since we have all the necessary tools o.board. So +1 for modularity through, flags, configs etc. We would need to decide what it should be though. Currently and previously the configuration basically happened in the switch case.
2. as noted above, it makes it even easier. We can improve though. 
3. The makefile around there is more verbose than having it in JS, since it only consist of variables that will be iterated over files, which is probably baggage from the early days. There is no real reason not to do in JS to gain more flexibility and maintainability.
4. There is no apparent advantage but also there is no real change in there. The advantage will be that I can branch for `'guides'` and use components `'html'` doesn't need. Namely We will need a custom component for the `_toc` since it will be nested.